### DISCUSSION TITLE
Driver took too long to build. This is likely an issue with the browser or driver

### DISCUSSION THREAD

**Discussion by alex270795** [SeleniumHQ/docker-selenium#2511] [Q&A]
Hello, We are using Selenium Grid V4 and run tests via selenium-ide-runner.
When we run a large number of tests, they become queued and wait to be executed. But when 30 seconds pass and the test has not started, we get an error on Timeout.
![388851272-f02aa16c-630a-42bf-aa1a-f3cac100bf6d](https://github.com/user-attachments/assets/8909f7b7-6e5b-42c4-a7f6-a455b4795490)

It can be passed by increasing timeout hard-coded in script webdriver.ts
![388851732-0241a57b-4931-4d49-97cb-820ce5455fe6](https://github.com/user-attachments/assets/abb05790-51b6-4f8e-b88b-9977af11729e)

How is it possible to avoid this error or is it possible to define this Timeout for WebDriver via ${variable} ?
Has anyone faced that kind of issues?

**Comment by portyu9** [comment]
Hi @alex270795 

This timeout is coming from **Selenium IDE / `side-runtime`**, not from docker-selenium's session queue.

The driver creation code has a hard-coded **30 second** timeout while waiting for:

```js
new webdriver.Builder().build()
```

So if Grid queues the new-session request for longer than 30 seconds, the runner gives up even though Grid may still be perfectly capable of servicing that request later.

There currently isn't a supported `${variable}` / environment variable that changes this particular 30-second timeout.

Also, the runner's normal `--timeout` option is not the same setting; that controls how long commands wait when locating elements.

Increasing Grid's session-request timeout will therefore **not solve this specific failure** because the client is aborting first.

Your practical options are:

1. **Keep session allocation below 30 seconds**
   - pre-scale/warm browser Nodes
   - maintain some spare Node capacity
   - reduce the initial burst of runner workers

2. **Reduce `selenium-side-runner` parallelism**, for example with a lower `--max-workers`, so fewer sessions enter the Grid queue simultaneously.

3. If waiting longer than 30 seconds is an expected part of your architecture, **patch/fork `side-runtime`** to make that timeout configurable, or use a WebDriver runner/client without that hard-coded driver-build deadline.

So your manual modification of `webdriver.ts` works because it changes the actual timeout responsible for the error; unfortunately that timeout is not currently exposed as a normal runner/Grid configuration variable.

Please mark this answer as accepted if it helped, thank you.
